### ISSUE TITLE
SystemConsole (aka NetDriver) appears to have broke

### ISSUE THREAD

**Issue by tig** [migueldeicaza/gui.cs#518]
I figured I'd write a Scenario to illustrate `Application.UseSystemConsole = true;`

I immediately discovered I had to change how `Scenario.Init, .Setup, and .Run` worked. Which I did.

Then I couldn't figure out why I was getting weird behavior and crashes.

So I modified `demo.c` to `Application.UseSystemConsole = true;` and ran it.

Crashed on startup.

I don't have time to look at this now, but from what I can see nobody's been testing `SystemConsole` for a while and it's quite broken...

**Comment by BDisp** [comment]
SystemConsole does not support the gui. It is as if you are writing characters directly on the console. So this just uses the OS's default console.

**Comment by tig** [comment]
Oh. I thought it worked for gui, but just didn't support mouse.

**Comment by BDisp** [comment]
I tested it once and nothing works, at least I got this impression that made me think right away, why use this option if it doesn't work lol

**Comment by tig** [comment]
To be clear, `UseSystemConsole` means the `NetConsole` will be used. Just so we're talking about the same thing:

```csharp
		static void Init (Func<Toplevel> topLevelFactory)
		{
			if (_initialized) return;

			var p = Environment.OSVersion.Platform;
			Mono.Terminal.IMainLoopDriver mainLoopDriver;

			if (UseSystemConsole) {
				mainLoopDriver = new Mono.Terminal.NetMainLoop ();
				Driver = new NetDriver ();
			} else if (p == PlatformID.Win32NT || p == PlatformID.Win32S || p == PlatformID.Win32Windows) {
				var windowsDriver = new WindowsDriver ();
				mainLoopDriver = windowsDriver;
				Driver = windowsDriver;
			} else {
				mainLoopDriver = new Mono.Terminal.UnixMainLoop ();
				Driver = new CursesDriver ();
			}
			Driver.Init (TerminalResized);
			MainLoop = new Mono.Terminal.MainLoop (mainLoopDriver);
			SynchronizationContext.SetSynchronizationContext (new MainLoopSyncContext (MainLoop));
			Top = topLevelFactory ();
			Current = Top;
			CurrentView = Top;
			_initialized = true;
		}
```

Here are related Issues:

- #370
- #48 
- #84 
- #418 

I'll stop now and get back to something important.

**Comment by BDisp** [comment]
Of course we are talking about the same thing. But at the time I tested NetMainLoop, it was not and is not as developed as WindowsDriver and CursesDriver, so as I couldn't get it to work properly, I gave up and never looked at it again. This observation of yours, perhaps will enliven some work on it, who knows, if it is worth it.

**Comment by tig** [comment]
I'm trying to get a list of bugs that must be fixed for 1.0. I don't think anyone uses SystemConsole and thus would recommend we don't mark this as 1.0. Any objections?
